### PR TITLE
TELCODOCS-1154 - release notes for RAN 4.13 new features and bugs

### PR DESCRIPTION
**Pull request by aireilly**
Release notes for RAN 4.13 new features and bugs.  

Version(s):
enterprise-4.13

Issue:
https://issues.redhat.com/browse/TELCODOCS-1154

Link to docs preview:
* [Tech Preview status](https://58179--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-13-release-notes.html#ocp-4-13-scalability-tech-preview)
* [New features](https://58179--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-13-release-notes.html#ocp-4-13-scalability-and-performance)
* [Known issues](https://58179--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-13-release-notes.html#ocp-4-13-ran-known-issues)

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: release_notes/ocp-4-13-release-notes.adoc [hunk 1]
```diff
@@ -314,6 +314,17 @@ The cluster network can be expanded to support the addition of nodes to the clus
 
 In {product-title} {product-version}, {lvms} is supported in dual-stack for IPv4 and IPv6 network environments. For more information, see xref:../networking/ovn_kubernetes_network_provider/converting-to-dual-stack.adoc#nw-dual-stack-convert_converting-to-dual-stack[Converting to a dual-stack cluster network].
 
+[id="ocp-4-13-storage-lvms-resource-reduction"]
+==== Resource reduction of {lvms}
+
+In {product-title} {product-version}, the CPU and memory usage of {lvms} is reduced. The container image sizes are also decreased and you can apply labels to LVM {lvms} resources to enable workload partitioning.
```
Comment (amolnar-rh) [inline release_notes/ocp-4-13-release-notes.adoc:320]: RN for [OCPVE-226](https://issues.redhat.com/browse/OCPVE-226)

@suleymanakbas91 Please take a look when you have the time.

File: release_notes/ocp-4-13-release-notes.adoc [hunk 2]
```diff
@@ -314,6 +314,17 @@ The cluster network can be expanded to support the addition of nodes to the clus
...
+
+[id="ocp-4-13-storage-lvms-multiple-storage-class-support"]
+==== Multiple storage class support for {lvms}
+
+In {product-title} {product-version}, you can use multiple storage classes on a single Logical Volume Manager cluster.
+For more information, see xref:../../../storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc#lvms-creating-lvms-cluster_logical-volume-manager-storage
```
[the remark this answers is not in the document]
Comment (suleymanakbas91) [reply release_notes/ocp-4-13-release-notes.adoc]: This one looks good.

File: release_notes/ocp-4-13-release-notes.adoc (hunk 1 above)
Comment (suleymanakbas91) [reply release_notes/ocp-4-13-release-notes.adoc]: `you can apply labels to LVM {lvms} resources to enable workload partitioning.` The necessary labels are already applied to lvms resources so we could change this to something like `the required labels are applied to lvms resources so you can enable workload partitioning`.

Comment (suleymanakbas91) [reply release_notes/ocp-4-13-release-notes.adoc]: Sounds good :)

File: release_notes/ocp-4-13-release-notes.adoc [hunk 3]
```diff
@@ -671,6 +667,37 @@ Now, during container image pre-caching, {cgu-operator} compares the available h
 
 For more information see xref:../scalability_and_performance/cnf-talm-for-cluster-upgrades.adoc#talo-precache-feature-image-filter_cnf-topology-aware-lifecycle-manager[Using the container image pre-cache filter].
 
+[id="ocp-4-13-ran-http-transport"]
+==== HTTP transport replaces AMQP for PTP and bare-metal events
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:671]: @aneeshkp @josephdrichard Can you LGTM on this please.

File: release_notes/ocp-4-13-release-notes.adoc [hunk 4]
```diff
@@ -671,6 +667,37 @@ Now, during container image pre-caching, {cgu-operator} compares the available h
...
+
+HTTP is now the default transport in the PTP and bare-metal events infrastructure.
+AMQ Interconnect is EOL from 30 June 2024.
+When you use HTTP transport for PTP and bare-metal events, you need to perist the events subscription in the cluster using a `PersistentVolume`.
+
+For more information, see xref:../networking/using-ptp.adoc#cnf-about-ptp-fast-event-notifications-framework_using-ptp[About the PTP fast event notifications framework].
+
+[id="ocp-4-13-ran-westport-channel-grandmaster"]
+==== Support for Intel E810 Westport Channel NIC as PTP grandmaster clock
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:680]: @aneeshkp @josephdrichard Can you LGTM on this please.

File: release_notes/ocp-4-13-release-notes.adoc [hunk 5]
```diff
@@ -1964,6 +2007,16 @@ Workaround: Add the mandatory `spec.git_provider.user` field for Bitbucket. In a
 
 * When you include more than three failure domains in the `ControlPlaneMachineSet` definition, the load balancing algorithm does not prioritize existing control plane machines. If you add a fourth failure domain that is alphabetically higher in precedence than the existing three failure domains to the definition, the fourth failure domain takes precedence over any existing failure domains. This behavior can apply rolling forward updates to a control plane machine. You can prevent this issue by setting existing in-use failure domains to a higher precedence than the new and unused failure domains. This action stabilizes each control plane machine during the course of adding more than three failure domains to the definition. (link:https://issues.redhat.com/browse/OCPBUGS-11968[*OCPBUGS-11968*])
 
+[id="ocp-4-13-ran-known-issues"]
+* If you specify an invalid subscription channel in the subscription policy that you use to perform a cluster upgrade, the {cgu-operator-first} indicates that the upgrade is successful right after {cgu-operator} enforces the policy because the `Subscription` resource remains in the `AtLatestKnown` state.
```
Comment (kcarmichael08) [inline release_notes/ocp-4-13-release-notes.adoc:2011]: suggestion: "right" = "immediately" ?

**Review by kcarmichael08 — COMMENTED** [review]
Couple of comments; otherwise, looks good to me!

File: release_notes/ocp-4-13-release-notes.adoc [hunk 6]
```diff
@@ -1968,6 +2011,16 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
 
 * The Cluster Network Operator (CNO) controller is monitoring more resources than it needs to. As a result, its reconciler is being triggered too frequently, causing a much higher rate of API requests than necessary. There is approximately 1 config map access request made every second. This increases the load on both the CNO and the `kube-apiserver`. (link:https://issues.redhat.com/browse/OCPBUGS-11565[*OCPBUGS-11565*])
 
+[id="ocp-4-13-ran-known-issues"]
+* If you specify an invalid subscription channel in the subscription policy that you use to perform a cluster upgrade, the {cgu-operator-first} indicates that the upgrade is successful immediately after {cgu-operator} enforces the policy because the `Subscription` resource remains in the `AtLatestKnown` state.
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:2015]: @Missxiaoguo Can you LGTM on this 4.13 known issue text please.

File: release_notes/ocp-4-13-release-notes.adoc [hunk 7]
```diff
@@ -1968,6 +2011,16 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
...
+(link:https://issues.redhat.com/browse/OCPBUGS-9239[*OCPBUGS-9239*])
+
+* `kdump` fails to generate the `vmcore` crash dump file on HPE Edgeline e920t servers that are running the real-time kernel with Intel E810 NIC and ice driver.
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:2018]: @snits Can you LGTM on this 4.13 known issue text please.

File: release_notes/ocp-4-13-release-notes.adoc [hunk 8]
```diff
@@ -1968,6 +2011,16 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
...
+(link:https://issues.redhat.com/browse/RHELPLAN-138236[*RHELPLAN-138236*])
+
+* In {ztp}, when you provision a managed cluster that contains more than a single node using a `SiteConfig` CR, disk partition fails when one or more nodes has a `diskPartition` resource configured in the `SiteConfig` CR.
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:2021]: @pixelsoccupied can you Can you LGTM on this 4.13 known issue text please.

Comment (pixelsoccupied) [reply release_notes/ocp-4-13-release-notes.adoc:2021]: LGMT thanks!

File: release_notes/ocp-4-13-release-notes.adoc (hunk 7 above)
Comment (aireilly) [reply release_notes/ocp-4-13-release-notes.adoc]: Updated, PTAL

File: release_notes/ocp-4-13-release-notes.adoc [hunk 9]
```diff
@@ -2092,6 +2135,24 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
...
+* If you specify an invalid subscription channel in the subscription policy that you use to perform a cluster upgrade, the {cgu-operator-first} indicates that the upgrade is successful immediately after {cgu-operator} enforces the policy because the `Subscription` resource remains in the `AtLatestKnown` state.
+(link:https://issues.redhat.com/browse/OCPBUGS-9239[*OCPBUGS-9239*])
+
+* `kdump` fails to generate the `vmcore` crash dump file on HPE Edgeline e920t and HPE ProLiant DL110 Gen10 servers with Intel E810 NIC and ice driver installed.
+(link:https://issues.redhat.com/browse/RHELPLAN-138236[*RHELPLAN-138236*])
+
+* In {ztp}, when you provision a managed cluster that contains more than a single node using a `SiteConfig` CR, disk partition fails when one or more nodes has a `diskPartition` resource configured in the `SiteConfig` CR.
+(link:https://issues.redhat.com/browse/OCPBUGS-9272[*OCPBUGS-9272*])
+
+* In clusters configured with PTP boundary clocks (T-BC) and deployed vDU applications, sometimes messages are not sent from the follower interface of the T-BC on the vDU for periods of up to 40 seconds. The rate or errors in the logs can vary. An example log error is below:
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:2148]: @orosp can you review/LGTM on this 4.13 known issue please.

File: release_notes/ocp-4-13-release-notes.adoc [hunk 10]
```diff
@@ -2092,6 +2135,24 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
...
+* If you specify an invalid subscription channel in the subscription policy that you use to perform a cluster upgrade, the {cgu-operator-first} indicates that the upgrade is successful immediately after {cgu-operator} enforces the policy because the `Subscription` resource remains in the `AtLatestKnown` state.
+(link:https://issues.redhat.com/browse/OCPBUGS-9239[*OCPBUGS-9239*])
+
+* `kdump` fails to generate the `vmcore` crash dump file on HPE Edgeline e920t and HPE ProLiant DL110 Gen10 servers with Intel E810 NIC and ice driver installed.
+(link:https://issues.redhat.com/browse/RHELPLAN-138236[*RHELPLAN-138236*])
+
+* In {ztp}, when you provision a managed cluster that contains more than a single node using a `SiteConfig` CR, disk partition fails when one or more nodes has a `diskPartition` resource configured in the `SiteConfig` CR.
+(link:https://issues.redhat.com/browse/OCPBUGS-9272[*OCPBUGS-9272*])
+
+* In clusters configured with PTP boundary clocks (T-BC) and deployed vDU applications, sometimes messages are not sent from the follower interface of the T-BC on the vDU for periods of up to 40 seconds. The rate or errors in the logs can vary. An example error log is below:
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:2148]: @orosp can you LGTM on this OCP 4.13 known issue please

Comment (aireilly) [reply release_notes/ocp-4-13-release-notes.adoc:2148]: LGTM from @orosp on slack

File: release_notes/ocp-4-13-release-notes.adoc [hunk 11]
```diff
@@ -2101,6 +2144,45 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
...
+[source,terminal]
+----
+2023-01-15T19:26:33.017221334+00:00 stdout F phc2sys[359186.957]: [ptp4l.0.config] nothing to synchronize
+----
+(link:https://issues.redhat.com/browse/RHELPLAN-145492[*RHELPLAN-145492*])
+
+* RFC2544 performance tests show that the `Max delay` value for a packet to traverse the network is over the minimum threshold. This regression is found in {product-title} 4.13 clusters running the Telco RAN DU profile.
+(link:https://issues.redhat.com/browse/OCPBUGS-13224[*OCPBUGS-13224*])
+
+* When you install a {sno} cluster using {ztp} and configure PTP and bare-metal events with HTTP transport, the `linuxptp-daemon` daemon pod intermittently fails and does not have the required `PersistentVolumeClaim` (`PVC`) CR applied.
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:2169]: @jzding Can you review this 4.13 known issue please?

File: release_notes/ocp-4-13-release-notes.adoc [hunk 12]
```diff
@@ -2101,6 +2144,45 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
...
++
+.Example output
+[source,terminal]
+----
+mount: /var/lib/kubelet/plugins/kubernetes.io/local-volume/mounts/local-pv-bc42d358: mount(2) system call failed: Structure needs cleaning.
+----
+To workaround the issue, delete the `cloud-event-proxy-store-storage-class-http-events` `PVC` CR and re-deploy the PTP Operator.
+(link:https://issues.redhat.com/browse/OCPBUGS-12358[*OCPBUGS-12358*])
+
+* `OSLAT` performance tests run on a {sno} cluster running {product-title} 4.13 show a max latency result greater than 20 microseconds.
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:2180]: @jlelli can you please LGTM on this 4.13 known issue

File: release_notes/ocp-4-13-release-notes.adoc (hunk 11 above)
Comment (jzding) [reply release_notes/ocp-4-13-release-notes.adoc:2169]: the `linuxptp-daemon` daemon pod intermittently fails to deploy.

**Review by jzding — APPROVED** [review]
/lgtm

File: release_notes/ocp-4-13-release-notes.adoc [hunk 13]
```diff
@@ -2427,6 +2470,45 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
 +
 Workaround: By using SSH to connect to the node that is the current leader of the vsphere-syncer process and restarting the vsphere-syncer container (using crictl), this issue can be mitigated and the driver successfully comes up. (link:https://issues.redhat.com/browse/OCPBUGS-13385[*OCPBUGS-13385*])
 
+[id="ocp-4-13-ran-known-issues"]
+* If you specify an invalid subscription channel in the subscription policy that you use to perform a cluster upgrade, the {cgu-operator-first} indicates that the upgrade is successful immediately after {cgu-operator} enforces the policy because the `Subscription` resource remains in the `AtLatestKnown` state.
+(link:https://issues.redhat.com/browse/OCPBUGS-9239[*OCPBUGS-9239*])
```
Comment (aireilly) [inline release_notes/ocp-4-13-release-notes.adoc:2475]: @yliu127 can you LGTM on this please?

Comment (yliu127) [reply release_notes/ocp-4-13-release-notes.adoc:2475]: lgtm

File: release_notes/ocp-4-13-release-notes.adoc [hunk 14]
```diff
@@ -2427,6 +2470,45 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
...
+----
+mount: /var/lib/kubelet/plugins/kubernetes.io/local-volume/mounts/local-pv-bc42d358: mount(2) system call failed: Structure needs cleaning.
+----
+To workaround the issue, delete the `cloud-event-proxy-store-storage-class-http-events` `PVC` CR and re-deploy the PTP Operator.
+(link:https://issues.redhat.com/browse/OCPBUGS-12358[*OCPBUGS-12358*])
+
+* `OSLAT` performance tests run on a {sno} cluster running {product-title} 4.13 show a max latency result greater than 20 microseconds.
+(link:https://issues.redhat.com/browse/RHELPLAN-155443[*RHELPLAN-155443*])
+
+* `cyclictest` performance tests run on a {sno} cluster running {product-title} 4.13 show a max cyclictest latency result greater than 20 microseconds.
```
Comment (abhatt-rh) [inline release_notes/ocp-4-13-release-notes.adoc:2510]: Per IBM Style:
> Do not start a sentence with a lowercase term; rewrite the sentence.

Further, the linked Jira issue points to `oslat` and not `cyclictest`. Based on [Measuring latency](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/cnf-performing-platform-verification-latency-tests.html#cnf-measuring-latency_cnf-latency-tests), I understand that these are different tests. Moreover, the preceding point addresses the issue for `oslat`. Kindly verify.

File: release_notes/ocp-4-13-release-notes.adoc [hunk 15]
```diff
@@ -2427,6 +2470,45 @@ Workaround: Manually or by using a daemon set, modify the file `/etc/cni/tuning/
...
++
+.Example output
+[source,terminal]
+----
+mount: /var/lib/kubelet/plugins/kubernetes.io/local-volume/mounts/local-pv-bc42d358: mount(2) system call failed: Structure needs cleaning.
+----
+To workaround the issue, delete the `cloud-event-proxy-store-storage-class-http-events` `PVC` CR and re-deploy the PTP Operator.
+(link:https://issues.redhat.com/browse/OCPBUGS-12358[*OCPBUGS-12358*])
+
+* `OSLAT` performance tests run on a {sno} cluster running {product-title} 4.13 show a max latency result greater than 20 microseconds.
```
Comment (abhatt-rh) [inline release_notes/ocp-4-13-release-notes.adoc:2507]: Per IBM Style:
> Do not start a sentence with a lowercase term; rewrite the sentence.

Consider the following:
```suggestion
* The `oslat` performance tests that run on a {sno} cluster running {product-title} 4.13 show a max latency result greater than 20 microseconds.
```

**Review by tliu2021 — COMMENTED** [review]
/lgtm

**Review by abhatt-rh — COMMENTED** [review]
Hi @aireilly 
Nice work on the RAN release notes!
I added a couple of comments for your consideration. It seemed as though some bits were not peer reviewed (I might be wrong). You may also need to update the preview for the _Known issues_ section because the last couple of updates in that section aren't showing up.

Also, this PR needs QE approval.

File: release_notes/ocp-4-13-release-notes.adoc [hunk 16]
```diff
@@ -1025,6 +1021,37 @@ Now, during container image pre-caching, {cgu-operator} compares the available h
...
+A `ContainerRuntimeConfig` CR that configures crun as the default container runtime has been added to the GitOps ZTP `ztp-site-generate` container.
+
+For optimal performance in clusters that you install with {ztp}, enable crun for control plane and worker nodes in {sno}, {3no}, and standard clusters alongside additional day-0 installation manifest CRs.
+
+For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.adoc#ztp-sno-du-configuring-crun-container-runtime_sno-configure-for-vdu[Configuring crun as the default container runtime].
+
+[id="ocp-4-13-etcd-overview"]
+==== Documentation enhancement: Overview of etcd is now available
+
+An overview of etcd, including the benefits it provides and how it works, is now available in the {product-title} documentation. As the primary data store for Kubernetes, etcd provides a reliable approach to cluster configuration and management on {product-title} through the etcd Operator. For more information, see xref:../architecture/control-plane.adoc#etcd-overview_control-plane[Overview of etcd].
```
[the remark this answers is not in the document]
Comment (aireilly) [reply release_notes/ocp-4-13-release-notes.adoc:1052]: It was added by a member of the Core team.

**Comment by josclark42** [comment]
OCPBUGS-9272 note regarding disk partitions in SiteConfig LGTM

File: release_notes/ocp-4-13-release-notes.adoc [hunk 17]
```diff
@@ -1402,7 +1429,7 @@ In the following tables, features are marked with the following statuses:
 [id="ocp-4-13-rhv-deprecations"]
 ==== Red Hat Virtualization (RHV) as a host platform for {product-title} will be deprecated
 
-Red Hat Virtualization (RHV) as a host platform for {product-title} is now deprecated, and will be removed in the next {product-title} release, currently planned as {product-title} 4.14. 
+Red Hat Virtualization (RHV) as a host platform for {product-title} is now deprecated, and will be removed in the next {product-title} release, currently planned as {product-title} 4.14.
```
[the remark this answers is not in the document]
Comment (aireilly) [reply release_notes/ocp-4-13-release-notes.adoc:1432]: Not going to change this one :)

File: release_notes/ocp-4-13-release-notes.adoc (hunk 14 above)
Comment (aireilly) [reply release_notes/ocp-4-13-release-notes.adoc:2510]: Updated

File: release_notes/ocp-4-13-release-notes.adoc [hunk 18]
```diff
@@ -1025,6 +1021,37 @@ Now, during container image pre-caching, {cgu-operator} compares the available h
...
+You can now configure the Intel E810 Westport Channel NIC as a PTP grandmaster clock by using the PTP Operator.
+PTP grandmaster clocks use `ts2phc` (time stamp 2 physical clock) for system clock and network time synchronization.
+
+For more information, see xref:../networking/using-ptp.adoc#configuring-linuxptp-services-as-grandmaster-clock_using-ptp[Configuring linuxptp services as a grandmaster clock].
+
+[id="ocp-413-ran-ztp-crun-container-runtime"]
+==== Configuring crun as the default container runtime for managed clusters in {ztp}
+A `ContainerRuntimeConfig` CR that configures crun as the default container runtime has been added to the GitOps ZTP `ztp-site-generate` container.
+
+For optimal performance in clusters that you install with {ztp}, enable crun for control plane and worker nodes in {sno}, {3no}, and standard clusters alongside additional day-0 installation manifest CRs.
```
[the remark this answers is not in the document]
Comment (aireilly) [reply release_notes/ocp-4-13-release-notes.adoc]: Created https://github.com/redhat-documentation/vale-at-red-hat/pull/529 thanks

[the remark this answers is not in the document]
Comment (abhatt-rh) [reply release_notes/ocp-4-13-release-notes.adoc]: > Created [redhat-documentation/vale-at-red-hat#529](https://github.com/redhat-documentation/vale-at-red-hat/pull/529) thanks
 
Ooh, I like this approach. I could create so many of these from my peer review week :D
Thank you!

**Comment by linuxqiao** [comment]
/lgtm

Both RHELPLAN-155443 and RHELPLAN-155460 still in investigation.

**Comment by yuying1987** [comment]
lgtm for RHELPLAN-13823

**Pull request merged**